sui_core/src/authority/authority_store.rs

```diff
@@ -645,7 +663,13 @@ impl<const ALL_OBJ_VER: bool, S: Eq + Serialize + for<'de> Deserialize<'de>>
         // Delete objects.
         // Wrapped objects need to be deleted as well because we can no longer track their
         // content nor use them directly.
-        write_batch = write_batch.delete_batch(&self.objects, deleted.iter().map(|(id, _)| *id))?;
+        for key in deleted.iter() {
+            write_batch = write_batch.delete_range(
```

Comment:
No, we really don't want to delete older versions.   This would destroy any older checkpoints or older versions intentionally kept for other reasons.  Instead, just delete the version in `deleted`, which are items which transactions intentionally want to delete:

    write_batch.delete_batch(&self.objects,
        deleted.iter().map(|(_, obj)| (obj.id(), obj.version())))

Reply:
This is insufficient - we need to be able to be able to answer the question "has this object been deleted".